packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/ListView/index.js

```diff
@@ -45,212 +42,124 @@ import {
-        } = await get('/admin/webhooks');
+  const QUERY_KEY = 'webhooks';
+  const {
+    isLoading: webhooksLoading,
+    data: webhooks,
+    error: webhooksError,
+    refetch: refetchWebhooks,
+  } = useQuery(QUERY_KEY, async () => {
+    const {
+      data: { data },
+    } = await get('/admin/webhooks');
 
-        if (isMounted.current) {
-          dispatch({
-            type: 'GET_DATA_SUCCEEDED',
-            data,
-          });
-          notifyStatus('webhooks have been loaded');
-        }
-      } catch (err) {
-        console.log(err);
+    return data;
+  });
 
-        if (isMounted.current) {
-          if (err.code !== 20) {
-            toggleNotification({
-              type: 'warning',
-              message: { id: 'notification.error' },
-            });
-          }
-          dispatch({
-            type: 'TOGGLE_LOADING',
-          });
-        }
-      }
-    };
+  useEffect(() => {
+    if (webhooksError) {
+      toggleNotification({
+        type: 'warning',
+        message: formatAPIError(webhooksError),
+      });
 
-    if (canRead) {
-      fetchWebHooks();
+      return;
     }
-  }, [canRead, get, notifyStatus, toggleNotification]);
-
-  const handleToggleModal = () => {
-    setShowModal((prev) => !prev);
-  };
-
-  const handleConfirmDelete = () => {
-    if (webhookToDelete) {
-      handleConfirmDeleteOne();
-    } else {
-      handleConfirmDeleteAll();
+    if (webhooks) {
+      notifyStatus(
+        formatMessage({
+          id: 'Settings.webhooks.list.loading.success',
+          defaultMessage: 'Webhooks have been loaded',
+        })
+      );
     }
-  };
+  }, [webhooks, webhooksError, toggleNotification, formatMessage, notifyStatus, formatAPIError]);
 
-  const handleConfirmDeleteOne = async () => {
-    try {
-      await del(`/admin/webhooks/${webhookToDelete}`);
-
-      dispatch({
-        type: 'WEBHOOK_DELETED',
-        index: getWebhookIndex(webhookToDelete),
+  const deleteMutation = useMutation(
+    async () => {
+      await post('/admin/webhooks/batch-delete', {
+        ids: webhooksToDelete,
       });
-    } catch (err) {
-      /**
-       * TODO: especially this.
-       */
-      if (err.code !== 20) {
+      setWebhooksToDelete([]);
+
+      setShowModal(false);
+    },
+    {
+      onError(error) {
         toggleNotification({
           type: 'warning',
-          message: { id: 'notification.error' },
+          message: formatAPIError(error),
         });
-      }
+      },
+      onSuccess() {
+        refetchWebhooks();
+      },
     }
-    setShowModal(false);
-  };
-
-  const handleConfirmDeleteAll = async () => {
-    const body = {
-      ids: webhooksToDelete,
-    };
+  );
 
-    try {
-      await post('/admin/webhooks/batch-delete', body);
+  const enabledMutation = useMutation(
+    async ({ isEnabled, id }) => {
+      const body = {
+        ...webhooks.find((webhook) => webhook.id === id),
+        isEnabled,
+      };
+      delete body.id;
```

Comment:
```suggestion
      const { id, ...webhook } = webhooks.find((webhook) => webhook.id === id) ?? {}
      const body = {
        ...webhook,
        isEnabled,
      };
```

Nitpick but avoids mutation 👍🏼